addons/purchase/models/purchase.py

```diff
@@ -140,6 +141,11 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
         purchase_order_ids = self._search(expression.AND([domain, args]), limit=limit, access_rights_uid=name_get_uid)
         return self.browse(purchase_order_ids).name_get()
 
+    @api.depends('date_order', 'currency_id', 'company_id')
```

Comment:
you forgot the company_id.currency_id

Reply:
@amoyaux 
I think we should not allow to change currency rate even if we change company currency for the older order
also i don't in which case we will change company currency after placing order.

I will add the currency on the depends but I do't know we should change it for the existing order or not.

Reply:
If you were able to modify the company currency at a moment in an other you won't have a correct report for all the PO before the modification. So it's more safe to add it in the depends.

Reply:
@amoyaux I don't know in which case we will **change company currency** ( it should be same as installed CoA )

in case we change the **currency of the PO** ( it will be impacted by the currency_rate ) not based on the change of company currency.

What if we have already confirmed orders?

Also this case with sale module too.

cc: @lap-odoo

Reply:
I know it's not a possible usecase, it's just a good practice to put in the depends the fields that impact the compute

Reply:
@amoyaux  I added fields in the depends and also remove digits from currency_rate.